Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -267,12 +247,7 @@ extension HTTPClient {
             guard let scheme = url.scheme?.lowercased() else {
                 throw HTTPClientError.emptyScheme
             }
-
-            self.kind = try Kind(forScheme: scheme)
-            self.host = try self.kind.hostFromURL(url)
-            self.socketPath = try self.kind.socketPathFromURL(url)
-            self.uri = self.kind.uriFromURL(url)
-
+            (self.kind, self.host, self.socketPath, self.uri) = try Self.deconstructURL(url)
```

Comment:
Great work! Thank you!

Would you mind returning the `scheme` and `port` in `deconstructURL(_:)` as well?
`scheme` is already a stored property in `HTTPClient.Request` but `port` not. 
https://github.com/swift-server/async-http-client/blob/5dfbba8bb569c0eb98b927d139e81644e344703e/Sources/AsyncHTTPClient/HTTPHandler.swift#L266-L268
However, I don't mind adding `port` as another stored property. This would allow us to better reuse this functionality in the upcoming new `HTTPClientRequest` for async/await support as described in this proposal: 
https://github.com/swift-server/async-http-client/blob/96872470475f18a5667fbf681e2e0f67ad8a0110/docs/async-await.md#new-httpclientrequest-type

Reply:
Friendly Ping 🙂
I would like to use your changes soon. Do you have any estimate when you have time to work on this again?

Reply:
@karwa friendly ping 🙂
I would like to use your changes soon. Do you have any estimate when you have time to work on this again?

Reply:
Sure, I'll address the comments now and update the patch